infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/replication/receive/LogEntryWriter.java

```diff
@@ -102,7 +106,14 @@ private boolean processMsg(LogReplicationEntryMsg txMessage) {
                         log.warn("Skip applying log entries for stream {} as it is noisy. LR could be undergoing a rolling upgrade", streamId);
                         continue;
                     }
-                    for (SMREntry smrEntry : opaqueEntry.getEntries().get(streamId)) {
+
+                    List<SMREntry> smrEntries = opaqueEntry.getEntries().get(streamId);
+                    if (streamId.equals(REGISTRY_TABLE_ID)) {
+                        // Only keep entries that currently not exist in registry table
```

Comment:
nit: Only retain tables that are not present in the registry table

Reply:
Changed, thx!